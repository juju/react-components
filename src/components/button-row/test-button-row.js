/* Copyright (C) 2017 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const ButtonRow = require('./button-row');

describe('ButtonRow', function() {
  const renderComponent = (options = {}) =>
    enzyme.shallow(<ButtonRow buttons={options.buttons || []} />);

  it('generates a button', function() {
    var callbackStub = sinon.stub();
    var buttons = [
      {
        title: 'My button',
        type: 'submit',
        action: callbackStub
      }
    ];
    const wrapper = renderComponent({buttons});
    expect(wrapper).toMatchSnapshot();
  });
});
