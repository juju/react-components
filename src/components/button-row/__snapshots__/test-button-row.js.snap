// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRow generates a button 1`] = `
<div
  className="button-row"
>
  <ul
    className="p-inline-list"
  >
    <li
      className="p-inline-list__item"
      key="0"
    >
      <Button
        type="submit"
      >
        My button
      </Button>
    </li>
  </ul>
</div>
`;

exports[`ButtonRow generates more than one button 1`] = `
<div
  className="button-row"
>
  <ul
    className="p-inline-list"
  >
    <li
      className="p-inline-list__item"
      key="0"
    >
      <Button
        type="submit"
      >
        My first button
      </Button>
    </li>
    <li
      className="p-inline-list__item"
      key="1"
    >
      <Button
        type="submit"
      >
        My second button
      </Button>
    </li>
  </ul>
</div>
`;
